### PR TITLE
Keepers - Remove source from Registrar

### PR DESCRIPTION
**Pull request by cmalec**

### CODE REVIEW

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Changes look good. Will need to fix tests too

File: contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol [hunk 1]
```diff
@@ -37,11 +37,12 @@ contract KeeperRegistrar2_0 is TypeAndVersionInterface, ConfirmedOwner, ERC677Re
 
   /**
    * @notice versions:
+   * - KeeperRegistrar 1.1.1: Remove source from register
    * - KeeperRegistrar 1.1.0: Add functionality for sender allowlist in auto approve
    *                        : Remove rate limit and add max allowed for auto approve
    * - KeeperRegistrar 1.0.0: initial release
    */
-  string public constant override typeAndVersion = "KeeperRegistrar 1.1.0";
+  string public constant override typeAndVersion = "KeeperRegistrar 1.1.1";
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol:45]: not sure how we should version it. I think if this breaks customer contract (which I think this change will?) we should make it 2.0

File: contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol [hunk 2]
```diff
@@ -412,12 +400,13 @@ contract KeeperRegistrar2_0 is TypeAndVersionInterface, ConfirmedOwner, ERC677Re
...
-  modifier isActualSender(address expected, bytes memory data) {
-    address actual;
-    assembly {
-      actual := mload(add(data, 292))
-    }
-    if (expected != actual) {
+  modifier isActualSender(address expected, bytes calldata data) {
+    // decode register function arguments to get actual sender
+    (, , , , , , , address sender) = abi.decode(
+      data[4:],
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol:406]: I believe `4` is to remove the function signature? In function above `permittedFunctionsForLINK` we add 32 bytes to data and then get selector. I'm not sure how both are working at the same time

File: contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol (hunk 1 above)
Comment (cmalec) [reply contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol:45]: why would this break customer contract? It only affects new registers. Is there other changes in here already other than mine?

File: contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol (hunk 2 above)
Comment (cmalec) [reply contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol:406]: add(X, 32) is for a `memory` string so the first 32 bytes contains the length and data follows. So add(X, 32) skips that and points to the data.
vs
The first four bytes of a contract call(`calldata`) is the function selector

File: contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol (hunk 1 above)
Comment (cmalec) [reply contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol]: breaks our example of programmatic upkeeps

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Can you also fix the assembly in `isActualAmount` please :)

**Review by infiloop2 — DISMISSED** [review]
looks good. thanks for simplifying the assembly!

**Review by infiloop2 — APPROVED** [review]

**Review by infiloop2 — DISMISSED** [review]

**Review by infiloop2 — DISMISSED** [review]

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**